### PR TITLE
Feature survey

### PR DESCRIPTION
**Pull request by chinmay10**

### CODE REVIEW

**Comment by sidbav** [comment]
I am merging this right now just to get it in for iteration 1; will go through code review afterwards.

File: app/views/home/index.html.erb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 <% name = current_user.present? ? current_user.first_name + ' ' + current_user.last_name : 'Stranger' %>
 
 <h1>Welcome, <%= name %> </h1>
-<p>Find me in app/views/home/index.html.erb</p>
\ No newline at end of file
+<% if current_user.present? %>
```
Comment (sidbav) [inline app/views/home/index.html.erb:4]: We should probably not have this screen at all, we should simply have login screen when the user first goes to the website.

File: app/views/surveys/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <td><%= survey.survey_category %></td>
+        <td><%= "Incomplete" %></td>
+        <td><%= link_to "Take Survey", select_language_survey_path(survey) %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+</div>
+
+<%= link_to "New survey", new_survey_path %>
```
Comment (sidbav) [inline app/views/surveys/index.html.erb:28]: This path should not be visible by default; anyone should not be able to create new surveys.

File: app/views/surveys/select_language.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h1>Select Language</h1>
+
+<ul>
+  <% @languages.each do |language| %>
+    <li><%= link_to language, start_survey_survey_path(@survey, language: language) %></li>
```
Comment (sidbav) [inline app/views/surveys/select_language.html.erb:5]: Language names should be written in the appropriate language. Spanish should be Español for example.

File: app/views/surveys/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<p style="color: green"><%= notice %></p>
```
Comment (sidbav) [inline app/views/surveys/show.html.erb:1]: This view will need to be hidden from regular users.

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,7 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+
+survey1 = Survey.create(survey_name: "U.S. Household Food Security Survey", survey_category: "Food Security", languages: "English, Spanish, Chinese")
```
Comment (sidbav) [inline db/seeds.rb:9]: languages is currently one string, not a list, should probably be updated to something this, or whatever ruby's syntax is for arrays: ["English", "Spanish", "Chinese"]

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,7 @@
...
+survey2 = Survey.create(survey_name: "Six-Item Short Form of the Food Security Survey", survey_category: "Food Security", languages: "English")
```
Comment (sidbav) [inline db/seeds.rb:10]: same thing here.

File: features/step_definitions/survey_steps.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  if page.has_link?("Sign Out")
+    click_link "Sign Out"
+  end
+end
+
+#pending
+Given(/^I am logged in$/) do
+  unless page.has_link?("Sign Out")
+    # code to log user in
+    user = User.find_by(email: "testuser@test.com")
```
Comment (sidbav) [inline features/step_definitions/survey_steps.rb:19]: Is this required?

File: spec/requests/questions_spec.rb [hunk 8]
```diff
@@ -0,0 +1,135 @@
+require 'rails_helper'
```
Comment (sidbav) [inline spec/requests/questions_spec.rb:1]: I am assuming this is auto generated via rails.. have you tried it out? What are the results?

File: spec/requests/surveys_spec.rb [hunk 9]
```diff
@@ -0,0 +1,135 @@
+require 'rails_helper'
```
Comment (sidbav) [inline spec/requests/surveys_spec.rb:1]: Same question here.

**Review by sidbav — COMMENTED** [review]
Some comments, suggestions and questions!

File: app/views/home/index.html.erb (hunk 1 above)
Comment (chinmay10) [reply app/views/home/index.html.erb:4]: okay so we need change the route definition for redirecting only signed users to home screen. (next iteration)

File: app/views/surveys/index.html.erb (hunk 2 above)
Comment (chinmay10) [reply app/views/surveys/index.html.erb:28]: Above route definition will solve this issue as well

File: app/views/surveys/select_language.html.erb (hunk 3 above)
Comment (chinmay10) [reply app/views/surveys/select_language.html.erb:5]: yes, Needs to be changed in seeded values

File: features/step_definitions/survey_steps.rb (hunk 7 above)
Comment (chinmay10) [reply features/step_definitions/survey_steps.rb:19]: no not required.

File: spec/requests/questions_spec.rb (hunk 8 above)
Comment (chinmay10) [reply spec/requests/questions_spec.rb:1]: auto generated

File: spec/requests/surveys_spec.rb (hunk 9 above)
Comment (chinmay10) [reply spec/requests/surveys_spec.rb:1]: auto generated

File: app/views/surveys/index.html.erb (hunk 2 above)
Comment (sidbav) [reply app/views/surveys/index.html.erb:28]: Dont think so, since we dont want regular patients to be able to create new surveys.

**Pull request merged**